hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieMultiTableDeltaStreamer.java

```diff
@@ -0,0 +1,259 @@
+      typedProperties.setProperty(Constants.TARGET_SCHEMA_REGISTRY_URL_PROP, schemaRegistryBaseUrl + typedProperties.getString(Constants.KAFKA_TOPIC_PROP) + schemaRegistrySuffix);
+    }
+  }
+
+  public static void main(String[] args) throws IOException {
+    JavaSparkContext jssc = UtilHelpers.buildSparkContext("multi-table-delta-streamer", Constants.LOCAL_SPARK_MASTER);
+    try {
+      new HoodieMultiTableDeltaStreamer(args, jssc).sync();
+    } finally {
+      jssc.stop();
+    }
+  }
+
+  private static String getCommonPropsFileName(String[] args) {
+    String commonPropsFileName = "common_props.properties";
+    for (int i = 0; i < args.length; i++) {
+      if (args[i].equals(Constants.PROPS_FILE_PROP)) {
+        commonPropsFileName = args[i + 1];
+        break;
+      }
+    }
+    return commonPropsFileName;
+  }
+
+  private static String getConfigFolder(String[] args) {
+    String configFolder = "";
+    for (int i = 0; i < args.length; i++) {
+      if (args[i].equals(Constants.CONFIG_FOLDER_PROP)) {
+        configFolder = args[i + 1];
+        break;
+      }
+    }
+    return configFolder;
+  }
+
+  /**
+   * Resets target table name and target path using base-path-prefix.
+   * @param args
+   * @param database
+   * @param tableName
+   * @return
+   */
+  private static String resetTarget(String[] args, String database, String tableName) {
+    int counter = 0;
+    String targetBasePath = "";
+    for (int i = 0; i < args.length; i++) {
+      if (args[i].equals(Constants.BASE_PATH_PREFIX_PROP)) {
+        args[i + 1] = args[i + 1].charAt(args[i + 1].length() - 1) == '/' ? args[i + 1].substring(0, args[i + 1].length() - 1) : args[i + 1];
+        targetBasePath = args[i + 1] + Constants.FILEDELIMITER + database + Constants.FILEDELIMITER + tableName;
+        counter += 1;
+      } else if (args[i].equals(Constants.TARGET_TABLE_ARG)) {
+        args[i + 1] = database + Constants.DELIMITER + tableName;
+        counter += 1;
+      }
+      if (counter == 2) {
+        break;
+      }
+    }
+
+    return targetBasePath;
+  }
+
+  /*
+  Creates actual HoodieDeltaStreamer objects for every table/topic and does incremental sync
+   */
+  public void sync() {
+    for (TableExecutionObject object : tableExecutionObjects) {
+      try {
+        new HoodieDeltaStreamer(object.getConfig(), jssc, object.getProperties()).sync();
+        successTables.add(object.getDatabase() + Constants.DELIMITER + object.getTableName());
+      } catch (Exception e) {
+        logger.error("error while running MultiTableDeltaStreamer for table: " + object.getTableName(), e);
+        failedTables.add(object.getDatabase() + Constants.DELIMITER + object.getTableName());
+      }
+    }
+
+    logger.info("Ingestion was successful for topics: " + successTables);
+    if (!failedTables.isEmpty()) {
+      logger.info("Ingestion failed for topics: " + failedTables);
+    }
+  }
+
+  public static class Constants {
```

Comment:
We would need to document these  properties and add a section for HoodieMultiDeltaStreamer just like https://hudi.incubator.apache.org/docs/writing_data.html#deltastreamer. This can be done as a followup.

Reply:
yeah will take it up next once we are done with code review of this PR.

Reply:
Filed jira HUDI-766 and assigned to you.

Reply:
Will take care of that :)